test/jdk/javax/swing/JFileChooser/4150029/bug4150029.java

```diff
@@ -21,73 +21,129 @@
+            System.out.println("Created directory: " + testDir);
+
+            subDir = new File(testDir, "subDir");
+            subDir.mkdir();
+            subDir.deleteOnExit();
+            System.out.println("Created sub-directory: " + subDir);
+
+            fileChooser = new JFileChooser(subDir);
+
+//            fileChooser.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
+
+            SwingUtilities.invokeAndWait(() -> {
+                frame = new JFrame("Backspace Shortcut for Directory Navigation Test");
+                frame.getContentPane().setLayout(new BorderLayout());
+                fileChooser = new JFileChooser(subDir);
+                fileChooser.setControlButtonsAreShown(false);
+                frame.getContentPane().add(fileChooser, BorderLayout.CENTER);
+                frame.pack();
+                frame.setLocationRelativeTo(null);
+                frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+                frame.setVisible(true);
+            });
+
+            doTesting();
+        } finally {
+            SwingUtilities.invokeAndWait(() -> {
+                if (frame != null) {
+                    frame.dispose();
+                }
+            });
         }
+    }
 
-        System.out.println("Temp directory: " + tmpDir);
-
-        File testDir = new File(tmpDir, "testDir");
-
-        testDir.mkdir();
-
-        File subDir = new File(testDir, "subDir");
-
-        subDir.mkdir();
-
-        System.out.println("Created directory: " + testDir);
-        System.out.println("Created sub-directory: " + subDir);
-
-        JFileChooser fileChooser = new JFileChooser(testDir);
+    private static void doTesting() {
+        Point p = frame.getLocationOnScreen();
+        robot.mouseMove(p.x + 200, p.y + 200);
+        robot.mousePress(InputEvent.BUTTON1_MASK);
 
-        fileChooser.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
+        boolean passed_1 = false;
+        boolean passed_2 = false;
+        robot.waitForIdle();
 
-        try {
-            res = fileChooser.showOpenDialog(this) != JFileChooser.APPROVE_OPTION ||
-                    testDir.getCanonicalPath().equals(fileChooser.getSelectedFile().getCanonicalPath());
-        } catch (IOException e) {
-            res = false;
+        // check backspace key at subDir level
+        clickBackSpace();
+        if (prevDir != crntDir) {
+            passed_1 = true;
+        }
 
-            e.printStackTrace();
+        // check if backspace key changes directory at root level
+        while (!fileChooser.getFileSystemView().isFileSystemRoot(prevDir)) {
+            clickBackSpace();
+            if (prevDir == crntDir) {
+                passed_2 = true;
+                break;
+            }
         }
 
-        try {
-            subDir.delete();
-            testDir.delete();
-        } catch (SecurityException e) {
-            e.printStackTrace();
+        if (passed_1 && passed_2) {
```

Comment:
I think no need to print "Test Passed", can check for not passed and throw the runtime exception inside if block.